internal/epoch/epoch_manager_test.go

```diff
@@ -393,26 +393,26 @@ func TestIndexEpochManager_NoCompactionInReadOnly(t *testing.T) {
 	// Use assert.Eventually here so we'll exit the test early instead of getting
 	// stuck until the timeout.
 	var (
-		loadedDone bool
-		loadedErr  error
+		loadedDone atomic.Bool
+		loadedErr  atomic.Value
 	)
 
 	go func() {
-		defer func() {
-			loadedDone = true
-		}()
+		if err := te2.mgr.Refresh(ctx); err != nil {
+			loadedErr.Store(err)
+		}
 
-		loadedErr = te2.mgr.Refresh(ctx)
 		te2.mgr.backgroundWork.Wait()
+		loadedDone.Store(true)
 	}()
 
-	if !assert.Eventually(t, func() bool { return loadedDone }, time.Second*5, time.Second) {
+	if !assert.Eventually(t, loadedDone.Load, time.Second*5, time.Second) {
 		// Return early so we don't report some odd failure on the error check below
 		// when we just never managed to initialize the epoch manager.
 		return
 	}
 
-	assert.NoError(t, loadedErr, "refreshing read-only index")
+	assert.Nil(t, loadedErr.Load(), "refreshing read-only index")
```

Comment:
technically, `loadedErr` does not need to be an `atomic.Value` since this read _"happens after" (and only) after `loadedDone` has been set to true and `loadedErr` is set before `loadedDone` is set. Nevertheless, making this an atomic value since there are no explicit locks and the ordering of the reads could change in the future and a race could be inadvertently introduced.